poppy/tests/test_core.py

```diff
@@ -498,6 +498,43 @@ def test_FITSOpticalElement(tempdir='./'):
     assert foe.opd_slice == 1
     assert np.allclose(foe.opd, rect_mask*1e-9)
 
+def test_OPD_in_waves_for_FITSOpticalElement():
+    pupil_radius = 1 * u.m
+    pupil = poppy.CircularAperture(radius=pupil_radius)
+    reference_wavelength = 1 * u.um
+    single_wave_1um_lens = poppy.ThinLens(
+        name='Defocus',
+        nwaves=1,
+        reference_wavelength=reference_wavelength,
+        radius=pupil_radius
+    )
+    osys = poppy.OpticalSystem(oversample=1, npix=1024)
+    osys.add_pupil(pupil)
+    osys.add_pupil(single_wave_1um_lens)
+    osys.add_detector(0.01 * u.arcsec / u.pixel, fov_pixels=3)
+    # We have applied 1 wave of defocus at 1 um, so verify the center
+    # has lower flux than at 2 um (it should be the 'hole' of the donut)
+    psf_1um = osys.calc_psf(reference_wavelength)
+    center_pixel_value = psf_1um[0].data[1,1]
+    psf_2um = osys.calc_psf(2 * reference_wavelength)
+    assert psf_2um[0].data[1,1] > psf_1um[0].data[1,1]
+    # Now, use the single_wave_1um_lens optic to make a
+    # wavelength-independent 1 wave defocus
+    lens_as_fits = single_wave_1um_lens.to_fits(what='opd', npix=1536)
+    lens_as_fits[0].header['BUNIT'] = 'radian'
+    lens_as_fits[0].data *= 2 * np.pi / reference_wavelength.to(u.m).value
+    thin_lens_wl_indep = poppy.FITSOpticalElement(opd=lens_as_fits, opdunits='radian')
+    central_pixels = []
+    for prefactor in (0.5, 1.0, 2.0):
+        osys = poppy.OpticalSystem(oversample=1, npix=1024)
+        osys.add_pupil(pupil)
+        osys.add_pupil(thin_lens_wl_indep)
+        osys.add_detector(prefactor * 0.01 * u.arcsec / u.pixel, fov_pixels=3)
+        psf = osys.calc_psf(wavelength=prefactor * u.um)
+        central_pixels.append(psf[0].data[1,1])
+        yc, xc = psf[0].data.shape[0] // 2, psf[0].data.shape[1] // 2
```

Comment:
But I'm not sure this line with `yc, xc` has any effect?

Reply:
Ah yeah, that's an artifact of when I wasn't doing a 3 pixel FOV